aspnetcore/security/authorization/resourcebased.md

```diff
@@ -62,102 +56,58 @@ Task<bool> AuthorizeAsync(ClaimsPrincipal user,
 
-```csharp
-public async Task<IActionResult> Edit(Guid documentId)
-{
-    Document document = documentRepository.Find(documentId);
-
-    if (document == null)
-    {
-        return new HttpNotFoundResult();
-    }
-
-    if ((await _authorizationService.AuthorizeAsync(User, document, "EditPolicy")).Succeeded)
-    {
-        return View(document);
-    }
-    else
-    {
-        return new ChallengeResult();
-    }
-}
-```
+> [!NOTE]
+> The following code samples assume authentication has already run and set the `User` property.
 
-## Writing a resource based handler
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
-Writing a handler for resource based authorization is not that much different to [writing a plain requirements handler](policies.md#security-authorization-policies-based-authorization-handler). You create a requirement, and then implement a handler for the requirement, specifying the requirement as before and also the resource type. For example, a handler which might accept a Document resource would look as follows:
+[!code-csharp[](resourcebased/samples/ResourceBasedAuthApp2/Pages/Document/Edit.cshtml.cs?name=snippet_DocumentEditHandler)]
 
-```csharp
-public class DocumentAuthorizationHandler : AuthorizationHandler<MyRequirement, Document>
-{
-    protected override Task HandleRequirementAsync(AuthorizationHandlerContext context,
-                                                MyRequirement requirement,
-                                                Document resource)
-    {
-        // Validate the requirement against the resource and identity.
-
-        return Task.CompletedTask;
-    }
-}
-```
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
-Don't forget you also need to register your handler in the `ConfigureServices` method:
+[!code-csharp[](resourcebased/samples/ResourceBasedAuthApp1/Controllers/DocumentController.cs?name=snippet_DocumentEditAction)]
 
-```csharp
-services.AddSingleton<IAuthorizationHandler, DocumentAuthorizationHandler>();
-```
+---
 
-### Operational Requirements
+## Write a resource-based handler
 
-If you are making decisions based on operations such as read, write, update and delete, you can use the `OperationAuthorizationRequirement` class in the `Microsoft.AspNetCore.Authorization.Infrastructure` namespace. This prebuilt requirement class enables you to write a single handler which has a parameterized operation name, rather than create individual classes for each operation. To use it, provide some operation names:
+Writing a handler for resource-based authorization isn't much different than [writing a plain requirements handler](xref:security/authorization/policies#security-authorization-policies-based-authorization-handler). After creating a custom requirement class, implement a requirement handler class. The handler class specifies both the requirement and resource type. For example, a handler utilizing a `SameAuthorRequirement` requirement and a `Document` resource looks as follows:
```

Comment:
~ After creating a custom requirement class, implement a requirement handler class.~
Create a custom requirement class and implement a requirement handler class.